meta-mender-core/recipes-mender/mender-client/files/mender-client-resize-data-part.sh.in

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+set -e
+
+# All sizes are in blocks of 512 bytes
+
+MENDER_STORAGE_DEVICE=@MENDER_STORAGE_DEVICE@
+mender_storage_device_base=$(basename ${MENDER_STORAGE_DEVICE})
+
+# 32 MB, overstimate here and should be close enough
+PARTITION_OVERHEAD=65536
+
+total_disk_size=$(cat /sys/block/${mender_storage_device_base}/size)
+calculated_size=0
+
+for part in $(find /sys/block/${mender_storage_device_base}/ -name ${mender_storage_device_base}* -exec basename {} \;); do
+    if [[ "${part}" == "${mender_storage_device_base}" ]]; then
```

Comment:
`[[` is not valid `sh` syntax. If this worked for you it's probably only because `sh` was symlinked to `bash`. Only `[` is valid in `sh`.

Reply:
`dash` of all things :)